### ISSUE TITLE
RFE: machinectl integration with qemu-static

### ISSUE THREAD

**Issue by npmccallum** [systemd/systemd#4668]
### Submission type

  - [ ] Bug report
  - [x] Request for enhancement (RFE)

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### Description of feature

It is possible to download raw images and run them:
```
# machinectl pull-raw --verify=no http://ftp.halifax.rwth-aachen.de/fedora/linux/releases/24/CloudImages/x86_64/images/Fedora-Cloud-Base-24-1.2.x86_64.raw.xz
# systemd-nspawn -M Fedora-Cloud-Base-24-1.2.x86_64.raw
```

However, this raw image may not be of a compatible architecture (for example, the image could be ppc64 or aarch64). If qemu-user-static (Fedora; something similar exists in Debian) is installed, however, you can simply copy the appropriate qemu helper into the container before starting it. This allows all run binaries to be emulated.

Basically, machinectl would need to detect if the raw image can be run natively. If not, it needs to determine which helper should be run (processing binfmt.d entries). Then it should copy the helper into the container before running it.

**Comment by poettering** [comment]
hmm, i am tempted to say that stuff like this should better stay outside of systemd. I mean, it's easy to make this work, all you need to do is replace systemd-nspawn@foobar.service for that image by a slightly modified version of the original, that adds the qemu invocation...

**Comment by npmccallum** [comment]
... except that the invocation doesn't happen _outside_ the container but _inside_ the container.

**Comment by fpqc** [comment]
@npmccallum Is there any reason to want two levels of isolation?  Your emulated image is already isolated, right?

**Comment by npmccallum** [comment]
@fpqc There aren't two levels of isolation.
